### ISSUE TITLE
Incorrect "E0000 did not find expected '-' indicator" when there is a comment "!Sub" text

### ISSUE THREAD

**Issue by siebrand** [aws-cloudformation/cfn-lint#2653]
### CloudFormation Lint Version

0.76.1

### What operating system are you using?

Windows (WSL Ubuntu)

### Describe the bug

This may be a parser bug. I don't know too much of the internals. When the reproduction template is linted the following error is displayed:

```
E0000 did not find expected '-' indicator
test.yaml:25:19
```

When the line with "# comment" is removed from the reproduction template, cfn-lint exits with return code 0.

### Expected behavior

No error. The template code is valid.

### Reproduction template

```yaml
AWSTemplateFormatVersion: 2010-09-09
Description: cfn-lint test

Resources:
  Launchtemplate:
    Type: AWS::EC2::LaunchTemplate
    Properties:
      LaunchTemplateData:
        UserData:
          Fn::Base64: !Sub |
            #!/bin/bash
            /opt/aws/bin/cfn-init -v -s ${AWS::StackName} -r LaunchTemplate --region ${AWS::Region}
    Metadata:
      AWS::CloudFormation::Init:
        configSets:
          default:
            - foo
        foo:
          files:
            /tmp/bar:
              content: !Sub
                - text
                  
                  # comment
                  more text ${SomeVariable}
                - { SomeVariable: !FindInMap [ Variables, foo, bar ] }
Mappings:
  Variables:
    foo:
      bar: foo
```

**Comment by kddejong** [comment]
This is just invalid yaml as indicated by the error message.  

From cloudformation you will get `Template format error: YAML not well-formed. (line 25, column 19)`

Try 
```yaml
              content: !Sub
                - |
                  text
                  
                  # comment
                  more text ${SomeVariable}
                - { SomeVariable: !FindInMap [ Variables, foo, bar ] }
```
